rules/opentitan.bzl

```diff
@@ -351,30 +356,88 @@ bin_to_spiflash_frames = rule(
     },
 )
 
+def _gen_sim_dv_logs_db_impl(ctx):
```

Comment:
The short answer is no. The information that must be extracted from an ELF to make this optimization possible is tailored both to the LOG message formats and our chip-level testbench itself. 

Specifically, at the lowest level, the [testbench](https://cs.opensource.google/opentitan/opentitan/+/master:hw/dv/sv/sw_logger_if/sw_logger_if.sv?q=extract%20logs&ss=opentitan%2Fopentitan) needs to understand two things:
1. how to format a string printed using `printf` (which is ultimately invoked by the `LOG*()` macros)
2. the address the message is written too.

With this information the testbench is able to [monitor writes](https://cs.opensource.google/opentitan/opentitan/+/master:hw/dv/sv/sw_logger_if/sw_logger_if.sv;drc=990b3d8ebd2281764e5af9fec3b46095b38076c3;l=121-140) to said memory addresses and extract log messages on the fly. This avoids having to rely on printing messages over the UART, which is **_extremely_** slow in simulation (think of the baud rate vs. the core clock freq).

Unfortunately, since every device program may use a varying number, type, and format of `LOG*()` macros throughout its code, features we very much need to support, the only way to extract the format and address of each is at compile time, with custom tooling. There is no generic tooling that can provide such, as this is very specific to our testbench setup.

The significant improvements in simulation runtimes makes this well worth the added complexity to a couple bazel macros, for which we only need to define once (the meson setup was worse, and invoked a similar custom rule in **_many_** places in the repo, which was hard to maintain). In fact, this optimization should really be done for the Verilator platform too, as SW running in a Verilator sim still logs messages over UART, but that will require additional changes to the Verilator testbench (which is a topic for separate set of PRs).

Reply:
I should say that this framework looks quite reasonable to me. We had something very similar in a previous company I worked in (and it worked really well).